onnx/onnx.in.proto

```diff
@@ -379,16 +383,24 @@ message ModelProto {
   // all stored `TrainingInfoProto.initialization`s with the associated
   // `TrainingInfoProto.initialization_binding`s.
   //
-  // To use GraphCall and Gradient operators to generate gradient of trainable
-  // tensors, the user should put all trainable tensors to three places.
-  //  - inference graph's input list.
-  //  - inference graph's initializer list.
-  //  - update_binding(s) of the stored TrainingInfoProto.
-  // In addition, when using GraphCall, the user should also explicitly
-  // specify all trainable tensors in the input list of the GraphCall operator.
-  //
   // If this field is empty, the training behavior of the model is undefined.
-  repeated TrainingInfoProto training_info = 21;
+  repeated TrainingInfoProto training_info = 20;
+
+  // The following denote mutable global variables visible in ModelProto.graph as all as
+  // all graphs in TrainingInfoProto. These are initialized to the values specified
+  // in the initializer below, and then updated by initializer-bindings and
+  // update-bindings.
+  //
+  // If an input or initializer in a graph has the same name as a global variable,
+  // then the local variable in the graph will hide the global variable.
+  //
+  // This field usually stores trainable tensors of the model, because trainable
+  // tensors are used by both inference graph and training algorithm graph.
+  repeated TensorProto global_initializer = 30;
+
+  // Sparse initializers which are globally accessible.
+  // See the document of `global_initializer` for details.
+  repeated SparseTensorProto global_sparse_initializer = 31;
```

Comment:
Merge these two lists into a string list? Each element is a name of `ModelProto.graph.initializer`.

Reply:
(a) I thought the above version also allowed initializers from the training-algorithm?
(b) I forget whether today's meeting said it is okay to explicitly specify this, or just infer it from update_bindings?
(c) We didn't discuss one point in meeting: whether this list allows any (ModelProto.graph) initializer or only initializers that appear in input?

Reply:
> (a) I thought the above version also allowed initializers from the training-algorithm?

If the mutable initializer is only needed in a training graph, its name won't appear in the global variable name list.

> (b) I forget whether today's meeting said it is okay to explicitly specify this, or just infer it from update_bindings?

Ok with both but it seems people like `update_bindings` more.

> (c) We didn't discuss one point in meeting: whether this list allows any (ModelProto.graph) initializer or only initializers that appear in input?

No because
1. We only need `initializers` (trainable and mutable tensors) for training. I don't feel an input is a mutable thing.
2. If an input is promoted to global, multiple graph calls will have to use the same global input, which sounds strange.